### PR TITLE
Add a session secret

### PR DESCRIPTION
**Pull request by LostKobrakai**
Does allow for invalidating singular sessions, without affecting others.

Regarding #24

### CODE REVIEW

File: lib/login/session.ex [hunk 1]
```diff
@@ -2,31 +2,37 @@ defmodule Doorman.Login.Session do
...
   """
   def get_current_user(conn) do
     id = Plug.Conn.get_session(conn, @session_key)
+    secret = Plug.Conn.get_session(conn, @session_secret)
     repo = Application.get_env(:doorman, :repo)
     user_module = Application.get_env(:doorman, :user_module)
 
-    if id != nil do
-      repo.get(user_module, id)
+    if not is_nil(id) and not is_nil(secret) do
```
Comment (BlakeWilliams) [inline lib/login/session.ex]: Can we use ! and `&&` here instead of `and` and `not`?

File: lib/auth/secret.ex [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    Changeset.put_change(changeset, :session_secret, random_string(length))
+  end
+
+  # Generate a secure random string
+  defp random_string(length) do
+    :crypto.strong_rand_bytes(length)
+    |> Base.url_encode64
+    |> binary_part(0, length)
+  end
+end
```
Comment (BlakeWilliams) [inline lib/auth/secret.ex]: I think you want to setup your editor to add a trailing newline to the end of files to keep git happy

**Review by BlakeWilliams — COMMENTED** [review]
This is an awesome change! I've added two minor comments blocking merging right now. I can get it later, but if you could get it in the meantime I can hit the merge button.

File: lib/auth/secret.ex (hunk 2 above)
Comment (LostKobrakai) [reply lib/auth/secret.ex]: I usually have an editorconfig to set that up.

**Comment by LostKobrakai** [comment]
I've updated the PR to your comments.

File: lib/auth/secret.ex (hunk 2 above)
Comment (BlakeWilliams) [reply lib/auth/secret.ex]: Good idea, I'd accept a PR for that as well.

**Comment by BlakeWilliams** [comment]
@LostKobrakai Awesome, this is a great change. I'll make a new release soon. 💯

**Pull request merged**